ngi_reports/utils/entities.py

```diff
@@ -491,169 +602,41 @@ def populate(self, log, organism_names, **kwargs):
                 )
                 continue
 
-            customer_name = sample.get("customer_name", "NA")
-            # Get once for a project
-            if self.dates["first_initial_qc_start_date"] is not None:
-                self.dates["first_initial_qc_start_date"] = sample.get(
-                    "first_initial_qc_start_date"
-                )
-
             log.info(f"Processing sample {sample_id}")
+
             # Check if the sample is aborted before processing
-            if sample.get("details", {}).get("status_(manual)") == "Aborted":
+            if sample_info.get("details", {}).get("status_(manual)") == "Aborted":
                 log.info(f"Sample {sample_id} is aborted, so skipping it")
-                self.aborted_samples[sample_id] = AbortedSampleInfo(
-                    customer_name, "Aborted"
+                self.aborted_samples[sample_id] = Sample(
+                    sample_id, sample_info, status="Aborted"
                 )
                 continue
-
-            samObj = Sample()
-            samObj.ngi_id = sample_id
-            samObj.customer_name = customer_name
-            samObj.well_location = sample.get("well_location")
-            # Basic fields from Project database
-            # Initial qc
-            if sample.get("initial_qc"):
-                for item in samObj.initial_qc:
-                    samObj.initial_qc[item] = sample["initial_qc"].get(item)
-                    if (
-                        item == "initial_qc_status"
-                        and sample["initial_qc"]["initial_qc_status"] == "UNKNOWN"
-                    ):
-                        samObj.initial_qc[item] = "NA"
-
-            # Library prep
-            # Get total reads if available or mark sample as not sequenced
-            try:
-                # Check if sample was sequenced. More accurate value will be calculated from flowcell yield
-                total_reads = float(sample["details"]["total_reads_(m)"])
-            except KeyError:
+            # Check if sample was sequenced. More accurate value will be calculated from flowcell yield.
+            if not sample_info.get("details", {}).get("total_reads_(m)"):
                 log.warning(
-                    f"Sample {sample_id} doesnt have total reads, so adding it to NOT sequenced samples list."
+                    f"Sample {sample_id} doesn't have total reads, "
+                    "adding it to NOT sequenced samples list."
                 )
-                self.aborted_samples[sample_id] = AbortedSampleInfo(
-                    customer_name, "Not sequenced"
+                self.aborted_samples[sample_id] = Sample(
+                    sample_id, sample_info, status="Not sequenced"
                 )
                 # Don't gather unnecessary information if not going to be looked up
                 if not kwargs.get("yield_from_fc"):
                     continue
+            # self.library_construction.lower()
```

Comment:
A comment to self?

Reply:
Hm... I do not remember how that got there 😅